src/Netscript/NetscriptHelpers.ts

```diff
@@ -248,20 +248,20 @@ function makeRuntimeErrorMsg(ctx: NetscriptContext, msg: string, type = "RUNTIME
   const stack = errstack.split("\n").slice(1);
   const ws = ctx.workerScript;
   const caller = ctx.functionPath;
-  const scripts = ws.getServer().scripts;
   const userstack = [];
   for (const stackline of stack) {
-    let filename;
-    for (const script of scripts) {
-      if (script.filename && stackline.includes(script.filename)) {
-        filename = script.filename;
-      }
-      for (const dependency of script.dependencies) {
-        if (stackline.includes(dependency.filename)) {
-          filename = dependency.filename;
-        }
+    function getFileName() {
+      // Filename is current file if url found
+      if (ws.scriptRef.url && stackline.includes(ws.scriptRef.url)) return ws.scriptRef.filename;
+      // Also check urls for dependencies
+      for (const [url, script] of ws.scriptRef.dependencies) if (stackline.includes(url)) return script.filename;
```

Comment:
Style: I'm personally allergic to one-line if statements, but I accept they're probably actually fine, especially for cases like early return. One-line `for`s are a bit too far IMO, especially chained as a for-if.